Add new openldap module.

leifmadsen opened the pull request:
New OpenLDAP module.

Signed-off-by: Leif Madsen leif@leifmadsen.com

leifmadsen committed:
Add new openldap module.

Signed-off-by: Leif Madsen <leif@leifmadsen.com>

pabelanger commented:
I see alot of hardcoded settings in params.pp, EG: passwords, with no way to override them.

leifmadsen commented:
I have now removed all the hardcoded stuff that needs to be overridden.

leifmadsen committed:
Update variable names in params.pp and templates to not have a leading openldap_ tag, and to define defaults.
Also added a README.md file that lists all the variables you should configure in your node manifest.

Signed-off-by: Leif Madsen <leif@leifmadsen.com>
